contrib/generate-emulation.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/python3
+#
+# Copyright (C) 2017 Richard Hughes <richard@hughsie.com>
+#
+# SPDX-License-Identifier: LGPL-2.1+
+#
+# pylint: disable=invalid-name,missing-docstring
+
+import json
+import sys
+
+from typing import Dict, List, Any
+
+
+def _minimize_json(json_str: str) -> str:
+    nodes = json.loads(json_str)
+    new_attrs: List[Dict[str, Any]] = []
+    new_devices: List[Dict[str, Any]] = []
+    try:
+        for attr in nodes["SecurityAttributes"]:
+            new_attr: Dict[str, Any] = {}
+            for key in attr:
+                if key in ["AppstreamId", "HsiResult", "Flags", "Plugin"]:
+                    new_attr[key] = attr[key]
+            new_attrs.append(new_attr)
+    except KeyError:
+        pass
+    try:
+        for device in nodes["Devices"]:
+            new_device: Dict[str, Any] = {}
+            for key in device:
+                if key not in ["Created", "Modified", "Releases", "Plugin"]:
+                    new_device[key] = device[key]
+            new_devices.append(new_device)
+    except KeyError:
+        pass
+    return json.dumps(
+        {"SecurityAttributes": new_attrs, "Devices": new_devices},
+        indent=2,
+        separators=(",", " : "),
+    )
+
+
+if len(sys.argv) < 2:
```

Comment:
I have an idea here... How about you use the client API with gobject introspection to fetch the data instead of piping or arguments? Then this script can be "totally standalone".

I'm thinking a bit bigger...
I could see us installing it "with" fwupd and then when we want to reproduce any problem not specific to actual writing we can ask someone to run it and give us the json file it spits out.

Reply:
Can that be phase 2? There are some other API changes we'd need to do (e.g. dropping the _`Protocol`, but `Protocols` if more than 1 value_ compatibility hacks to do this in a clean way.

Reply:
Yeah; then it's trivial changes to this script as the daemon revs and does other API changes.

Reply:
Added.